src/BlockTypes/Checkout.php

```diff
@@ -100,6 +98,21 @@ protected function enqueue_data( array $attributes = [] ) {
 			$this->hydrate_from_api( $data_registry );
 			$this->hydrate_customer_payment_methods( $data_registry );
 		}
+
+		Package::container()->get( PaymentMethodAssets::class )->register_payment_method_data();
+	}
+
+	/**
+	 * Register/enqueue scripts used for this block.
+	 *
+	 * @param array $attributes  Any attributes that currently are available from the block.
+	 *                           Note, this will be empty in the editor context when the block is
+	 *                           not in the post content on editor load.
+	 */
+	protected function enqueue_scripts( array $attributes = [] ) {
+		$dependencies = Package::container()->get( PaymentMethodAssets::class )->get_all_registered_payment_method_script_handles();
+
+		Assets::register_block_script( $this->block_name . '-frontend', $this->block_name . '-block-frontend', $dependencies );
```

Comment:
This is where we'll need to put back in `woocommerce_blocks_enqueue_checkout_block_scripts_after` so the tracking hooked in on jetpack recently isn't broken.

Reply:
lol we're breaking compatibility before we even ship it 🚢 I hate hooks.

Reply:
> I hate hooks.

I share your sentiment.